python/models/model_builder.py

```diff
@@ -0,0 +1,101 @@
+from transformers import AutoModel, AutoTokenizer, AutoConfig
+from abc import ABC, abstractmethod
+import torch
+import shark_turbine.aot as aot
+
+
+class ModelBuilder(ABC):
+    """
+    Abstract base class for building compiled turbine modules.
+    """
+    def __init__(self, example_input:torch.Tensor = None) -> None:
+        super().__init__()
+        self.example_input = example_input
+        self.compiled_binary = None
+
+    def build_model(self) -> torch.nn.Module:
+        """
+        Abstract method for building a PyTorch model.
+        """
+        self.example_input = None
+        self.model = None
+        pass
+
+    def get_compiled_module(self, save_to: str = None) -> aot.CompiledModule:
+        """
+        Compiles the PyTorch model into a compiled module using SHARK-Turbine's AOT compiler.
+
+        Args:
+            save_to (str): one of: input (Torch IR) or import (linalg).
+
+        Returns:
+            aot.CompiledModule: The compiled module binary.
+        """
+        if self.compiled_binary is None:
+            #TODO: check cloud storage for existing ir
+            module = aot.export(self.build_model(), self.example_input)
+            self.compiled_binary = module.compile(save_to=save_to)
+        return self.compiled_binary
+    
+            
+    def infer(self) -> None:
```

Comment:
I'd leave this out for now. Ideally you will want a separate file that implements the inferencer behind this since it can ultimately not have the dependency bloat that this one will.